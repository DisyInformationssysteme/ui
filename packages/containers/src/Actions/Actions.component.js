import React, { PropTypes } from 'react';
import { api } from '@talend/react-cmf';
import { Actions as PureActions } from '@talend/react-components';

/**
 * @param {Object} props react props
 * @example
<Actions name="Hello world"></Actions>
 */
function Actions({ names, ...rest }, context) {
	const onClick = (event, payload) => {
		context.store.dispatch(payload.action.payload);
	};
	const actions = names ? names.map(
		name => api.action.getActionInfo(context, name),
	) : null;

	return (
		<PureActions
			actions={actions}
			{...rest}
			onClick={onClick}
		/>
	);
}

Actions.propTypes = {
	names: PropTypes.arrayOf(PropTypes.string),
};

Actions.contextTypes = {
	store: React.PropTypes.object,
};

Actions.displayName = 'CMF(Actions)';

export default Actions;
