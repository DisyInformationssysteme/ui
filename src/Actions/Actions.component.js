import React, { PropTypes } from 'react';
import { ButtonGroup } from 'react-bootstrap';
import Action from './Action';
import ActionDropdown from './ActionDropdown';

const TYPE_ACTION = 'action';
const TYPE_DROPDOWN = 'dropdown';

function getButtonGroupProps(props) {
	const buttonGroupProps = {};
	Object.keys(ButtonGroup.propTypes).forEach((id) => {
		if (props[id] !== undefined) {
			buttonGroupProps[id] = props[id];
		}
	});
	return buttonGroupProps;
}

/**
 * @param {object} props react props
 * @example
const actions: [
    {
		type: 'action',
		label: 'edit',
		icon: 'fa fa-edit',
		onClick: action('onEdit'),
	},
    {
		type: 'action',
		label: 'delete',
		icon: 'fa fa-trash-o',
		onClick: action('onDelete'),
	},
    {
		type: 'dropdown',
		label: 'related items',
		icon: 'fa fa-file-excel-o',
		items: [
			{
				label: 'document 1',
				onClick: action('document 1 click'),
			},
			{
				label: 'document 2',
				onClick: action('document 2 click'),
			},
		],
	}
 ];
<Actions actions={actions} tooltipPlacement="right" hideLabel link />
 */
function Actions(props) {
	const buttonGroupProps = getButtonGroupProps(props);
	return (
		<ButtonGroup className="tc-actions" {...buttonGroupProps}>
			{props.actions.map((action, index) => {
				const { type, ...rest } = action;
				switch (type) {
				case TYPE_ACTION:
					return (
						<Action
							hideLabel={props.hideLabel}
							key={index}
							link={props.link}
							tooltipPlacement={props.tooltipPlacement}
							{...rest}
						/>
					);
				case TYPE_DROPDOWN:
					return (
						<ActionDropdown
							hideLabel={props.hideLabel}
							key={index}
							link={props.link}
							tooltipPlacement={props.tooltipPlacement}
							{...rest}
						/>
					);
				default:
					return null;
				}
			})}
		</ButtonGroup>
	);
}

Actions.propTypes = {
	actions: PropTypes.arrayOf(
		PropTypes.oneOfType([
			PropTypes.shape({
				type: PropTypes.oneOf([TYPE_ACTION, TYPE_DROPDOWN]),
				...Action.propTypes,
			}),
			PropTypes.shape({
				type: PropTypes.oneOf([TYPE_ACTION, TYPE_DROPDOWN]),
				...ActionDropdown.propTypes,
			}),
		])
	),
	hideLabel: PropTypes.bool,
	tooltipPlacement: Action.propTypes.tooltipPlacement,
	link: PropTypes.bool,
	...ButtonGroup.propTypes,
};

Actions.defaultProps = {
	actions: [],
};

export default Actions;
