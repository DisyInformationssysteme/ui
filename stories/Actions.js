import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { Actions } from '../src/index';

const actions = [
	{ label: 'Preparations', icon: 'fa fa-asterisk', onClick: action('Preparations clicked'), bsStyle: 'primary' },
	{ label: 'Datasets', icon: 'fa fa-file-excel-o', onClick: action('Datasets clicked') },
	{ label: 'Favorites', icon: 'fa fa-star', onClick: action('Favorites clicked') },
];
storiesOf('Actions', module)
	.addWithInfo('default', () => (
		<div>
			<h1>Actions</h1>
			<h2>Definition</h2>
			<p>The actions component display buttons to let the user dispatch those actions</p>
			<h2>Examples</h2>
			<p>By default :</p>
			<Actions actions={actions} />
			<p>With hideLabel option</p>
			<Actions
				actions={actions}
				hideLabel
			/>
			<p>Vertical</p>
			<Actions
				actions={actions}
				vertical
			/>
			<p>Vertical with hideLabel</p>
			<Actions
				actions={actions}
				placement="right"
				hideLabel
				vertical
			/>
		</div>
	));
