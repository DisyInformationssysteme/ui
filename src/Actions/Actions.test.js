import React from 'react';
import renderer from 'react-test-renderer';

import Actions from './Actions.component';

jest.mock('react-dom');

describe('Actions', () => {
	it('should render buttons', () => {
		const actions = [
			{
				label: 'Preparations',
				icon: 'fa fa-asterisk',
				onClick: jest.fn(),
				bsStyle: 'primary',
			},
			{
				label: 'Datasets',
				icon: 'fa fa-file-excel-o',
				onClick: jest.fn(),
			},
			{
				label: 'Favorites',
				icon: 'fa fa-star',
				onClick: jest.fn(),
			},
		];
		const wrapper = renderer.create(<Actions actions={actions} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
