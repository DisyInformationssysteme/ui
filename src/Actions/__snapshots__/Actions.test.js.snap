exports[`Actions should render buttons 1`] = `
<div
  className="tc-actions btn-group">
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    role={null}
    type="button">
    <i
      className="fa fa-asterisk" />
    <span>
      Preparations
    </span>
  </button>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    role={null}
    type="button">
    <i
      className="fa fa-file-excel-o" />
    <span>
      Datasets
    </span>
  </button>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    role={null}
    type="button">
    <i
      className="fa fa-star" />
    <span>
      Favorites
    </span>
  </button>
</div>
`;
